swarm/network/stream/stream.go

```diff
@@ -47,6 +47,31 @@ const (
 	HashSize         = 32
 )
 
+//Enumerate options for syncing and retrieval
+type SyncingOption int
+type RetrievalOption int
+
+//Syncing options
+const (
+	//Syncing disabled
+	SyncingDisabled SyncingOption = iota
+	//Register the client and the server but not subscribe
+	SyncingRegisterOnly
+	//Both client and server funcs are registered, subscribe sent automatically
+	SyncingAutoSubscribe
+)
+
+const (
+	//Retrieval disabled
+	RetrievalDisabled RetrievalOption = iota
```

Comment:
Please explain why would retrieval be disabled?

Reply:
Added a short desc (disabling is mostly relevant for tests, in order to isolate and test syncing functionality)